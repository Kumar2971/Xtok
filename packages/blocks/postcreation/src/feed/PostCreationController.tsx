import PostCreationCommonController from './PostCreationCommonController'

// Customizable Area Start
import ImagePicker from "react-native-image-crop-picker";
import { getStorageData } from "../../../../framework/src/Utilities";
import { Alert } from 'react-native';
// Customizable Area End

export const configJSON = require("./config");

export default class PostCreationController extends PostCreationCommonController {
  // Customizable Area Start
  // Customizable Area End

  async componentDidMount() {
    super.componentDidMount();
    // Customizable Area Start
    const language = await getStorageData("SelectedLng");

    this.setState({
       mediadetails:this.props.route?.params?.mediadetails,
       language:language
    })

if(this.props.route?.params?.description){
  this.setState({caption:this.props.route?.params?.description ?? ''})
}

   if(this.props.route?.params?.poster ?? ''){
  this.setState({resultPoster:this.props.route?.params?.poster ?? ''})
   }

   if(this.props.route?.params?.bucketdetails){
    this.setState({bucketDetails:this.props.route?.params?.bucketdetails})
   }

   if(this.props.route?.params?.audioData){
    this.setState({audioData:this.props.route?.params?.audioData});
   }

    // Customizable Area End
  }
  // Customizable Area Start

  chooseImage = () => {
    ImagePicker.openPicker({
      multiple: false,
      mediaType: "photo",
      compressImageQuality: 0.3,
      includeBase64: true,
      cropping: true,
    }).then(async (image:any) => {
      let filename = image.path.substring(
        image.path.lastIndexOf("/") + 1,
        image.path.length
      );
      this.setState({
        image: image.sourceURL,
        profileImageData: {
          data: image.data,
          filename: filename,
          content_type: image.mime
        }
      });
    });
  };
  // Customizable Area End
}
