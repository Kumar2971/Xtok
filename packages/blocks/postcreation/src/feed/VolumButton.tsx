import React, { useState} from 'react';
// Customizable Area Start
import {
  StyleSheet,
  TouchableOpacity,
  Animated,
  Dimensions
} from 'react-native';
import Icon from "react-native-vector-icons/FontAwesome5"
const {height, width} = Dimensions.get('window');
//@ts-ignore
const VolumeButton = ({isMute, setisMute}) => {

  const [viewAnim] = useState(new Animated.Value(0));
 
  const onVolumePress = () => {
    setisMute(!isMute);
  
    Animated.timing(viewAnim, {
      toValue: 1,
      duration: 500,
      useNativeDriver: true,
    }).start(() => {
      fadeOut();
    });
  };

  const fadeOut = () => {
    Animated.timing(viewAnim, {
      delay: 500,
      toValue: 0,
      duration: 300,
      useNativeDriver: true,
    }).start();
  };

  return (
    <TouchableOpacity
      testID='btnID'
      activeOpacity={1}
      onPress={onVolumePress}
      style={styles.touchArea}>
      <Animated.View
        style={[
          styles.muteBtn,
          {
            opacity: viewAnim,
          },
        ]}>
            {(isMute &&  <Icon name="volume-mute" size={16} color="white"  style={styles.mute}/> ) ||  <Icon name="volume-up" size={16} color="white"  style={styles.mute}/>  }
          
         
   
      </Animated.View>
    </TouchableOpacity>
  );
};

export {VolumeButton};
const styles = StyleSheet.create({
  outer: {
    paddingHorizontal: 20,
    paddingVertical: 15,
    borderRadius: 5,
    borderWidth: 1,
    marginVertical: 5,
    minWidth: 100,
    alignItems: 'center',
    justifyContent: 'center',
  },
  dot: {
    fontSize: 16,
    alignItems: 'center',
    justifyContent: 'center',
    fontWeight: 'bold',
  },
  followText: {
    fontSize: 13,
    paddingRight: 10,
    paddingLeft: 2,
    paddingVertical: 8,
    fontWeight: 'bold',
  },
  touchArea: {
    width,
    height: height,
    position: 'absolute',
    justifyContent: 'center',
    alignItems: 'center',
  },
  muteBtn: {
    position: 'absolute',
    height: 60,
    width: 60,
    borderRadius: 30,
    backgroundColor: 'rgba(0,0,0,0.5)',
    justifyContent: 'center',
    alignItems: 'center',
  },
  mute: {
    height: 20,
    width: 20,
  },
});
// Customizable Area Start