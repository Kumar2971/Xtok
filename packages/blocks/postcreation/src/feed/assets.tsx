export const edit = require("../../assets/edit.png");
export const mainIcon = require("../../assets/main-icon.png");
export const effects = require("../../assets/effects.png");
export const gallery = require("../../assets/gallery.png");
export const crossicon = require("../../assets/crossicon.png");
export const beautify = require("../../assets/beautify.png");
export const filters = require("../../assets/filters.png");
export const flash = require("../../assets/flash.png");
export const flip = require("../../assets/flip.png");
export const speed = require("../../assets/speed.png");
export const timer = require("../../assets/timer.png");
export const sounds = require("../../assets/sounds.png");
export const Check = require("../../assets/Check.png");
export const yellowcircle = require("../../assets/yellowcircle.png");
export const ArrowLeft = require("../../assets/ArrowLeft.png");
export const AutoLayoutHorizontal = require("../../assets/AutoLayoutHorizontal.png");
export const Visibility = require("../../assets/Show.png");
export const Comments = require("../../assets/Chat.png");
export const Schedule = require("../../assets/Calendar.png");
export const Audience = require("../../assets/audience.png");
export const ArrowRight = require("../../assets/ArrowRight.png");
export const pausebutton = require("../../assets/pausebutton.png");
export const flasFilled= require("../../assets/flash-filled.png")
export const timerFilled= require("../../assets/timer-filled.png")

export const timeRing = require("../../assets/time_ring.png")
export const backArrow = require("../../../../blocks/global_assets/backArrow.png");
export const RightArrowfull = require("../../../../blocks/global_assets/RightArrowfull.png");


export const burningEffect = require('../../assets/burning_effect.png');
export const vikingHelmet = require('../../assets/viking_helmet.png');
export const vendetta = require('../../assets/vendetta_mask.png');
export const stallone = require('../../assets/stallone.png');
export const pingPong = require('../../assets/ping_pong.png');
export const humanoid = require('../../assets/humanoid.png');
export const fireEffect = require('../../assets/fire_effect.png');
export const emotionExagerator = require('../../assets/emotions_exaggerator.png');
// Customizable Area Start
export const avatar = require('../../assets/avatar.jpeg');
export const SwitchOff = require("../../../../blocks/global_assets/SwitchOff.png");
export const SwitchOn = require("../../../../blocks/global_assets/SwitchOn.png");
// Customizable Area End