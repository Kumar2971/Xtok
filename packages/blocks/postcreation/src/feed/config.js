Object.defineProperty(exports, "__esModule", {
  value: true
});

// Customizable Area Start
exports.validationApiContentType = "application/json";
exports.validationApiMethodType = "GET";
exports.GetApiMethodType = "GET";
exports.PostAPiMethod = "POST";
exports.exampleApiContentType = "application/json";
exports.PostApiMethodType = "GET";
exports.patchPostAPiMethod = "PATCH";
exports.postGetUrl = "posts/posts";
exports.postContentType = "application/json";
exports.deletePostAPiMethod = "DELETE";
exports.CategoryGetUrl = "categories/categories";
exports.btnExampleTitle = "CLICK ME";
exports.getAllCatergoryEndPoint = "categories/categories";
exports.getallconfigurationEndpoint="bx_block_posts/posts/get_post_configurations";
exports.searchAccount = "bx_block_elasticsearch/mentions_account_search";
exports.contentTypeApiUpdateUser = 'application/json';
exports.draftsListPutMethodType = 'PUT';
exports.draftsListDeleteEndPoint = 'bx_block_posts/posts/';
exports.draftsListUpdateEndPoint = 'bx_block_posts/posts/';
exports.searchHashTags = 'bx_block_elasticsearch/hashtag_search';
exports.autocompleteAPIKey ='AIzaSyA2reMhEodWUjV9o-WAWMF-N_8XSbZCox0'
// Customizable Area End
