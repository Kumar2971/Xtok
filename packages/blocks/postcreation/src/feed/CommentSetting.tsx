import React from "react";

// Customizable Area Start
import scale, { verticalScale } from "../../../../components/src/Scale";
import {
  View,
  Text,
  SafeAreaView,
  StyleSheet,
  Image,
  Dimensions,
  TouchableOpacity,
  BackHandler
} from "react-native";
let screenWidth = Dimensions.get("window").width;
let screenHeight = Dimensions.get("window").height;
const { height, width } = Dimensions.get("window");
import { mainIcon, ArrowLeft, yellowcircle, ArrowRight } from "../feed/assets";
// Customizable Area End

import PostCreationController from "./PostCreationController";
import { Props } from "./PostCreationCommonController";
import { translate } from '../../../../components/src/i18n/translate'
export default class CommentSetting extends PostCreationController {
  constructor(props: Props) {
    super(props);
  }
  // Customizable Area Start
  async componentDidMount() {
    super.componentDidMount();
    console.log("checking2", this.props.route?.params);
    this.setState({
      setcomments: this.props.route?.params?.setcomments,
    }, () => console.log("checking3", this.state.setcomments))
    BackHandler.addEventListener('hardwareBackPress', this.handleBackButtonClick);
    this.props.navigation.addListener('gestureEnd', this.handleBackButtonClick);
  }
  componentWillUnmount = async () => {
    BackHandler.removeEventListener('hardwareBackPress', this.handleBackButtonClick);
    this.props.navigation.addListener('gestureEnd', this.handleBackButtonClick);
  }

  handleBackButtonClick = () => {

    this.props.navigation.navigate("Postsubmit", {
      setcomments: this.state.setcomments,
      mediadetails: this.props.route?.params?.mediadetails,
                poster:this.props.route?.params?.poster,
                description: this.props.route?.params?.description
    });

    return true;
  }
  renderItem = () => {
    if (this.state.allCategories != "") {
      return (
        <View style={{ margin: 10 }}>
          <Text style={[styles.headerTxt,styles.headerText_Ar]}>
            {translate("Choose_setting_comments_for_this_video")}
          </Text>

          <View
            style={{
              display: "flex",
              alignItems: "center",
              flexDirection: "row",
              margin: 5,
            }}
          >
            <TouchableOpacity
             testID="setCommentBtnID"
              onPress={() => {
                this.setState({
                  setcomments: 0,
                });
              }}
            >
              {this.state.setcomments == 0 ? (
                <Image source={mainIcon} style={{ width: 30, height: 30 }} />
              ) : (
                <Image
                  source={yellowcircle}
                  style={{ width: 30, height: 30 }}
                />
              )}
            </TouchableOpacity>
            <View style={{ margin: 10 }}>
              <Text style={styles.headerTxt}>{translate("allow")}</Text>
            </View>
          </View>
          <View
            style={{
              display: "flex",
              alignItems: "center",
              flexDirection: "row",
              margin: 5,
            }}
          >
            <TouchableOpacity
            testID="setCommentBtnID1"
              onPress={() => {
                this.setState({
                  setcomments: 1,
                });
              }}
            >
              {this.state.setcomments == 1 ? (
                <Image source={mainIcon} style={{ width: 30, height: 30 }} />
              ) : (
                <Image
                  source={yellowcircle}
                  style={{ width: 30, height: 30 }}
                />
              )}
            </TouchableOpacity>
            <View style={{ margin: 10 }}>
              <Text style={styles.headerTxt}>{translate("hold_all")}</Text>
            </View>
          </View>
          <View
            style={{
              display: "flex",
              alignItems: "center",
              flexDirection: "row",
              margin: 5,
            }}
          >
            <TouchableOpacity
             testID="setCommentBtnID2"
              onPress={() => {
                this.setState({
                  setcomments: 2,
                });
              }}
            >
              {this.state.setcomments == 2 ? (
                <Image source={mainIcon} style={{ width: 30, height: 30 }} />
              ) : (
                <Image
                  source={yellowcircle}
                  style={{ width: 30, height: 30 }}
                />
              )}
            </TouchableOpacity>
            <View style={{ margin: 10 }}>
              <Text style={styles.headerTxt}>{translate("disable_comments")} </Text>
            </View>
          </View>
        </View>
      );
    }
  };
  // Customizable Area End
  
  render() {
    // Customizable Area Start
    // Customizable Area End
    return (
      <SafeAreaView style={styles.container}>
        {/* Customizable Area Start */}
        <View
          style={{
            // position: "absolute",
            width: "100%",
            paddingTop: 16,
          }}
        >
          <TouchableOpacity
            testID="goBack"
            onPress={() => {
              this.props.navigation.navigate("Postsubmit", {
                setcomments: this.state.setcomments,
                mediadetails: this.props.route?.params?.mediadetails,
                poster:this.props.route?.params?.poster,
                description: this.props.route?.params?.description
              });
            }}
          >
            <View
              style={{
                display: "flex",
                alignItems: "center",
                flexDirection: "row",
              }}
            >
              <Image source={
                this.state.language == 'ar' ? ArrowRight : ArrowLeft

              } style={styles.detailIcon} />
              <Text style={{ fontSize: scale(18), fontWeight: "bold", color: "#000000" }}>{translate("comments")}</Text>
            </View>
          </TouchableOpacity>
        </View>
        <View style={styles.bodyContent}>{this.renderItem()}</View>

        {/* Customizable Area End */}
      </SafeAreaView>
    );
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1,
    padding: 16
  },

  bodyContent: {
    flex: 1,
    height: screenHeight,
    flexDirection: "column",
    //s backgroundColor: "black",
    borderTopLeftRadius: 10,
    borderTopRightRadius: 10,
    //marginTop: 50,
  },
  pickImage: {
    width: screenWidth,
    height: screenWidth,
  },
  textStyleTitle: {
    fontSize: scale(18.7),
    marginStart: scale(17.7),
    marginTop: verticalScale(12.7),
  },
  headerTxt: {
    fontSize: scale(16),
    marginLeft: 10,
    flexWrap: 'wrap',
  },
  headerText_Ar: {
    marginBottom: scale(25),
    textAlign: "left"
  },

  crossBtn: {
    width: 30,
    height: 30,
    resizeMode: "contain",
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    position: "absolute",
    zIndex: 1000,
    top: 0,
    width: "100%",
  },
  textStylePrice: {
    fontSize: scale(18.2),
    marginStart: scale(17.7),
    marginTop: verticalScale(8.7),
  },

  row: {
    flexDirection: "row",
    height: scale(37.7),
    alignItems: "center",
    marginTop: verticalScale(14),
    paddingStart: scale(17.7),
  },

  viewEditDelete: {
    alignItems: "center",
    alignSelf: "center",
    top: 0,
    right: 0,
    position: "absolute",
    justifyContent: "flex-end",
    marginEnd: scale(17),
  },

  textActive: {
    alignSelf: "flex-start",
    paddingStart: scale(15),
    paddingEnd: scale(15),
    paddingTop: verticalScale(2),
    paddingBottom: verticalScale(2),
    borderRadius: 10,
    fontSize: scale(10.7),
    marginStart: scale(7),
    marginEnd: scale(7),
  },

  textPublish: {
    fontSize: scale(14.2),
  },

  textDescriptionTitle: {
    fontSize: scale(20),
    marginStart: scale(17.7),
    marginTop: verticalScale(5),
    fontWeight: "bold",
  },

  textDescription: {
    fontSize: scale(16),
    marginStart: scale(16.7),
    marginTop: verticalScale(8),
    textAlign: "left",
    // lineHeight: scale(20.7),
  },

  textPostId: {
    fontSize: scale(16.2),
    marginLeft: scale(16.7),
    marginVertical: verticalScale(5),
  },

  ButtonConatiner: {
    position: "absolute",
    bottom: verticalScale(14.3),
    left: 0,
    right: 0,
    shadowColor: "#000",
    shadowOpacity: 0.02,
    height: scale(72),
    width: scale(414),
    alignContent: "center",
  },

  InnerConatiner: {
    justifyContent: "space-between",
    flexDirection: "row",
    marginTop: verticalScale(14.3),
    marginHorizontal: scale(16.3),
  },

  customTxtStyle: {
    alignSelf: "center",
    fontSize: scale(16.5),
  },

  customTxtStyleClose: {
    alignSelf: "center",
    fontSize: scale(16.5),
  },

  buttonCustom: {
    width: scale(182.3),
    height: scale(43.3),
    borderRadius: 22.1,
    alignItems: "center",
    justifyContent: "center",
  },

  buttonCustomClose: {
    width: scale(182.3),
    height: scale(43.3),
    borderRadius: 22.1,
    borderWidth: 1,
    alignItems: "center",
    justifyContent: "center",
  },

  userConatiner: {
    height: verticalScale(66),
    borderTopWidth: 0.5,
    borderBottomWidth: 0.5,
    marginTop: verticalScale(13),
  },

  userProfileName: {
    flexDirection: "row",
    marginTop: verticalScale(14),
    alignItems: "center",
  },

  userImagesIcon: {
    width: scale(37.7),
    height: scale(38),
    borderRadius: scale(18.7),
  },

  userIcon: {
    width: scale(20),
    height: scale(24),
    alignSelf: "center",
    marginTop: verticalScale(7),
  },

  userName: {
    fontSize: scale(14.2),
    textAlign: "left",
  },

  userMemberShip: {
    fontSize: scale(10.7),
    textAlign: "left",
  },

  offerPrice: {
    marginBottom: verticalScale(12),
    marginTop: verticalScale(28),
    fontSize: scale(16.7),
  },

  buttonCustomSend: {
    width: scale(185.7),
    height: scale(43.3),
    borderRadius: 22.1,
    alignItems: "center",
    alignSelf: "center",
    justifyContent: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    elevation: 1,
  },

  containerBoxStyle: {
    flexGrow: 1,
    justifyContent: "center",
    paddingBottom: verticalScale(10),
  },
  detailIcon: {
    width: scale(30),
    height: scale(30),
    marginHorizontal: scale(15),
    resizeMode: "contain",
  },
});
// Customizable Area End
