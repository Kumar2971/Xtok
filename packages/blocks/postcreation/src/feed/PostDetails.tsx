import React from "react";

// Customizable Area Start
import scale, { verticalScale } from "../../../../components/src/Scale";
import {
  View,
  Text,
  SafeAreaView,
  StyleSheet,
  Image,
  Dimensions,
  TouchableOpacity,
  Platform,
} from "react-native";
import Video from 'react-native-video';
let screenWidth = Dimensions.get("window").width;
let screenHeight = Dimensions.get("window").height;
const { height, width } = Dimensions.get('window');
import { VolumeButton } from './VolumButton';
import Icon from "react-native-vector-icons/AntDesign";
import * as Progress from 'react-native-progress';
import CustomButton from "../../../../components/src/Custombutton";
// Customizable Area End

import PostCreationController from "./PostCreationController";
import { Props } from "./PostCreationCommonController";
import { translate } from "../../../../components/src/i18n/translate";
import { RightArrowfull } from "./assets";
import { getStorageData } from "../../../../framework/src/Utilities";
import { rightArrow } from "../../../helpcentre/src/assets";
export default class PostDetails extends PostCreationController {
  constructor(props: Props) {
    super(props);
    //@ts-ignore

  }
  async componentDidMount() {
    super.componentDidMount();
    // Customizable Area Start
    const language = await getStorageData("SelectedLng");
    this.setState({ language: language })
    this.props.navigation.addListener('focus', () => {
      this.setState({
        paused: false
      })
    })
  }

  render() {
    // Customizable Area Start
    // Customizable Area End
    return (
      <SafeAreaView style={styles.container}>
        {/* Customizable Area Start */}
        <View style={styles.bodyContent}>
        <View style={{flex:1}}>
                <View style={styles.headerVdo}>
                  {/* <View style={styles.progressbar}>
                    <Progress.Bar progress={this.state.progress} width={screenWidth - Scale(40)} unfilledColor={"white"} color={"#FFC925"} borderWidth={0} />
                  </View> */}
                  <TouchableOpacity testID="gobaack"
                    style={{ padding: 10 }}
                    onPress={() => {
                      this.props.navigation.goBack()
                      this.setState({
                        mediadetails: "",
                      })
                    }
                    }>
                    <Icon
                      name={this.state.language == 'ar' ? "arrowright" : "arrowleft"}
                      size={30}
                      style={styles.crossBtn}
                      color="white"
                    />
                  </TouchableOpacity>


                </View>
                <Video
                  ref={this.videoRef}
                  fullscreenAutorotate={true}
                  source={{ uri: this.state.mediadetails?.uri }}
                  // controls={true}
                  // onProgress={this.handleOnProgress}
                  // autoPlay={true}
                  repeat={true}
                  resizeMode={'contain'}
                  muted={this.state.isMute}
                  style={{ width: screenWidth, height: screenHeight, flex: 1 }}
                  playInBackground={false}
                  paused={this.state.paused}
                  // controls={true}
                //  ignoreSilentSwitch={'ignore'}
                // onLoadStart={onLoadStart}
                //   onLoad={onLoad}
                // onBuffer={videoBuffer}
                />

                <View style={styles.footerbuttons}>
                  <CustomButton testID="nextBtn" title={translate("next")} onPress={() => { this.props.navigation.navigate("Postsubmit",{mediadetails:this.state.mediadetails,bucketDetails:this.state.bucketDetails,poster:this.state.resultPoster,audioData: this.props?.route?.params?.audioData })
                      this.setState({ paused: true })}} />
                </View>
              </View>


        </View>

        {/* Customizable Area End */}
      </SafeAreaView>
    );
  }
}

// Customizable Area Start
const styles = StyleSheet.create({
  container: {
    flex: 1
  },

  bodyContent: {
    flex: 1,
    height: screenHeight,
    flexDirection: "column",
    backgroundColor: "black",
    // borderTopLeftRadius: 10,
    // borderTopRightRadius: 10,
  },
  pickImage: {
    width: screenWidth,
    height: screenWidth,

  },
  textStyleTitle: {
    fontSize: scale(18.7),
    marginStart: scale(17.7),
    marginTop: verticalScale(12.7)
  },
  headerTxt: { fontSize: scale(16), fontWeight: "bold" },
  crossBtn: {
    width: 30,
    height: 30,
    resizeMode: "contain"
  },
  header: {
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center", position: "absolute",
    zIndex: 1000,
    top: 0,
    width: "100%"
  },
  headerVdo: {
    position: "absolute",
    zIndex: 1000,
    top: 3,
    width: "100%"
  },
  footerbuttons: {
    paddingHorizontal: scale(20),
    flexDirection: "row",
    justifyContent: "flex-end",
    alignItems: "center",
    position: "absolute",
    zIndex: 1000,
    bottom: 60,
    width: "100%"
  },
  progressbar: {
    marginHorizontal: scale(20),
  },
  textStylePrice: {
    fontSize: scale(18.2),
    marginStart: scale(17.7),
    marginTop: verticalScale(8.7)
  },

  row: {
    flexDirection: "row",
    height: scale(37.7),
    alignItems: "center",
    marginTop: verticalScale(14),
    paddingStart: scale(17.7)
  },

  viewEditDelete: {
    alignItems: "center",
    alignSelf: "center",
    top: 0,
    right: 0,
    position: "absolute",
    justifyContent: "flex-end",
    marginEnd: scale(17)
  },

  textActive: {
    alignSelf: "flex-start",
    paddingStart: scale(15),
    paddingEnd: scale(15),
    paddingTop: verticalScale(2),
    paddingBottom: verticalScale(2),
    borderRadius: 10,
    fontSize: scale(10.7),
    marginStart: scale(7),
    marginEnd: scale(7)
  },

  textPublish: {
    fontSize: scale(14.2)
  },

  textDescriptionTitle: {
    fontSize: scale(20),
    marginStart: scale(17.7),
    marginTop: verticalScale(5),
    fontWeight: "bold"
  },

  textDescription: {
    fontSize: scale(16),
    marginStart: scale(16.7),
    marginTop: verticalScale(8),
    textAlign: "left"
  },

  textPostId: {
    fontSize: scale(16.2),
    marginLeft: scale(16.7),
    marginVertical: verticalScale(5)
  },

  ButtonConatiner: {
    position: "absolute",
    bottom: verticalScale(14.3),
    left: 0,
    right: 0,
    shadowColor: "#000",
    shadowOpacity: 0.02,
    height: scale(72),
    width: scale(414),
    alignContent: "center"
  },

  InnerConatiner: {
    justifyContent: "space-between",
    flexDirection: "row",
    marginTop: verticalScale(14.3),
    marginHorizontal: scale(16.3)
  },

  customTxtStyle: {
    alignSelf: "center",
    fontSize: scale(16.5)
  },

  customTxtStyleClose: {
    alignSelf: "center",
    fontSize: scale(16.5)
  },

  buttonCustom: {
    width: scale(182.3),
    height: scale(43.3),
    borderRadius: 22.1,
    alignItems: "center",
    justifyContent: "center"
  },

  buttonCustomClose: {
    width: scale(182.3),
    height: scale(43.3),
    borderRadius: 22.1,
    borderWidth: 1,
    alignItems: "center",
    justifyContent: "center"
  },

  userConatiner: {
    height: verticalScale(66),
    borderTopWidth: 0.5,
    borderBottomWidth: 0.5,
    marginTop: verticalScale(13)
  },

  userProfileName: {
    flexDirection: "row",
    marginTop: verticalScale(14),
    alignItems: "center"
  },

  userImagesIcon: {
    width: scale(37.7),
    height: scale(38),
    borderRadius: scale(18.7)
  },

  userIcon: {
    width: scale(20),
    height: scale(24),
    alignSelf: "center",
    marginTop: verticalScale(7)
  },

  userName: {
    fontSize: scale(14.2),
    textAlign: "left"
  },

  userMemberShip: {
    fontSize: scale(10.7),
    textAlign: "left"
  },

  offerPrice: {
    marginBottom: verticalScale(12),
    marginTop: verticalScale(28),
    fontSize: scale(16.7)
  },

  buttonCustomSend: {
    width: scale(185.7),
    height: scale(43.3),
    borderRadius: 22.1,
    alignItems: "center",
    alignSelf: "center",
    justifyContent: "center",
    shadowColor: "#000",
    shadowOffset: {
      width: 0,
      height: 1
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    elevation: 1
  },

  containerBoxStyle: {
    flexGrow: 1,
    justifyContent: "center",
    paddingBottom: verticalScale(10)
  }
});
// Customizable Area End
