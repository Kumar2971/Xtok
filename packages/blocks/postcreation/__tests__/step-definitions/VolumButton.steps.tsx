/// <reference types="@types/jest" />

import { defineFeature, loadFeature } from "jest-cucumber";
import { shallow, ShallowWrapper } from "enzyme";

import * as helpers from "../../../../framework/src/Helpers";
import { runEngine } from "../../../../framework/src/RunEngine";
import { Message } from "../../../../framework/src/Message";
import * as utils from "../../../../framework/src/Utilities";

import MessageEnum, {
  getName
} from "../../../../framework/src/Messages/MessageEnum";
import React from "react";
import {VolumeButton} from "../../src/feed/VolumButton";

const mockOnpress = jest.fn()
const screenProps = {
  isMute: false,
  setisMute: mockOnpress 
};

const mockgetStorageData = jest.spyOn(utils, "getStorageData")
const mockFetch = jest.fn();

const feature = loadFeature("./__tests__/features/VolumButton-scenario.feature");

defineFeature(feature, (test) => {
  beforeEach(() => {
  });
  jest.useFakeTimers();

  test("User navigates to VolumeButton", ({ given, when, then }) => {
    let exampleBlockA: ShallowWrapper;


    given("I am a User loading VolumeButton", () => {
      exampleBlockA = shallow(<VolumeButton {...screenProps} />);
    });

    when("I navigate to the VolumeButton", () => {
      expect(exampleBlockA).toMatchSnapshot()
    });

    then("VolumeButton will load with out errors",async () => {
      expect(exampleBlockA).toBeTruthy();
    });
    
    then("I can select mute button with out errors", () => {
      let nextBtn = exampleBlockA.findWhere(
        (node) => node.prop("testID") === "btnID"
      );
      nextBtn.simulate("press");
  
      expect(mockOnpress).toHaveBeenCalled();
    });
  });
});