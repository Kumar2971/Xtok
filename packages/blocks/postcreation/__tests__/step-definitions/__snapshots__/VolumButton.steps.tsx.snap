// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VolumeButton User navigates to VolumeButton 1`] = `
<ForwardRef
  activeOpacity={1}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "height": 1334,
      "justifyContent": "center",
      "position": "absolute",
      "width": 750,
    }
  }
  testID="btnID"
>
  <ForwardRef(AnimatedComponentWrapper)
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(0,0,0,0.5)",
          "borderRadius": 30,
          "height": 60,
          "justifyContent": "center",
          "position": "absolute",
          "width": 60,
        },
        Object {
          "opacity": 0,
        },
      ]
    }
  >
    <mockFontAwesome5
      color="white"
      name="volume-up"
      size={16}
      style={
        Object {
          "height": 20,
          "width": 20,
        }
      }
    />
  </ForwardRef(AnimatedComponentWrapper)>
</ForwardRef>
`;
